### DISCUSSION TITLE
Custom event in custom components ?

### DISCUSSION THREAD

**Discussion by loloof64** [DioxusLabs/dioxus#359] [Q&A] [answered]
Is it possible to define a custom event for a component ?

I mean, if I define a Board component, can I dispatch an "signal" named "Won" ?

```rust
#[derive(PartialEq, Props)]
pub struct BoardProps {
    cols_count: u8,
}

#[allow(non_snake_case)]
pub fn Board<'a>(cx: Scope<'a, BoardProps>) -> Element {
   cx.render(rsx!(
        div {
            class: "board",
            // Is there something for dispatching an event and its parameters so that a parent component can 
            // listen to "Won" event ?
            onclick: move |_| dispatch('Won', {white: true}); 
       }
  }
}
```

**Comment by jkelleyrtp** [comment, marked as answer]
While there's not really a way of "emitting" into parent events, you can just pass a handler in through BoardProps called "on_won" and then call it with whatever event you want.

https://dioxuslabs.com/guide/components/component_children.html#passing-handlers

I think this solves your use case - you want handlers above to be passed into components.

**Reply by loloof64** [reply]
Thank you very much :)

It works :)

**Reply by loloof64** [reply]
I used

```rust
#[derive(Props)]
pub struct BoardProps<'a> {
    cols_count: u8,
    onwon: EventHandler<'a>
}
```

and

```rust
cx.props.onwon.call(());
```

As I didn't need to pass data to the event.
